language-plutus-core/src/Language/PlutusCore/TypeCheck.hs

```diff
@@ -63,7 +63,7 @@ dynamicBuiltinNameMeaningsToTypes
     => ann -> DynamicBuiltinNameMeanings -> m DynamicBuiltinNameTypes
 dynamicBuiltinNameMeaningsToTypes ann (DynamicBuiltinNameMeanings means) = do
     let getType mean = do
-            ty <- liftQuote $ dynamicBuiltinNameMeaningToType mean
+            let ty = dynamicBuiltinNameMeaningToType mean
```

Comment:
Shouldn't there a be a `rename` here? Since we *are* running in `Quote`, and this is a closed-but-not-necessarily-globally-unique thing?

Reply:
I believe no, because I think *kind* checking does not require global uniqueness, because we do not do any type normalization and environments correctly reflect scopes. This needs to be documented, though.

I do plan to add `newtype` wrappers to stress where renaming is necessary and where is not.